program/rust/src/rust_oracle.rs

```diff
@@ -44,27 +56,138 @@ use crate::utils::pyth_assert;
 
 use super::c_entrypoint_wrapper;
 
+const PRICE_T_SIZE: usize = size_of::<pc_price_t>();
+const PRICE_ACCOUNT_SIZE: usize = size_of::<PriceAccountWrapper>();
+
+
 ///Calls the c oracle update_price, and updates the Time Machine if needed
 pub fn update_price(
     _program_id: &Pubkey,
-    _accounts: &[AccountInfo],
+    accounts: &[AccountInfo],
     _instruction_data: &[u8],
     input: *mut u8,
 ) -> OracleResult {
-    //For now, we did not change the behavior of this. this is just to show the proposed structure
-    // of the program
-    c_entrypoint_wrapper(input)
+    let price_account_info = &accounts[1];
+    let account_len = price_account_info.try_data_len()?;
+    if account_len < PRICE_ACCOUNT_SIZE {
+        if account_len != PRICE_T_SIZE {
+            return Err(ProgramError::InvalidArgument);
+        }
+        msg!("Please resize the account to allow for SMA tracking!");
```

Comment:
can we standardize on SMA or TWAP instead of using both terms

Reply:
(I prefer SMA since we're using EMA already)

Reply:
I do not believe I am using TWAP in my part, do you main rename things in the C- code?
I do recall having a discussion with Jeff, and him having a strong prefrence for "SMA"

Reply:
ok i thought i saw TWAP somewhere else in the pr. Definitely no need to go around renaming extra stuff as part of this.

(ah, your PR description says TwapTracker)

Reply:
also suggest not logging a message here, as it will look like an error in all the solana program logs. possible bad publicity if we release this.